idf_build_apps/app.py

```diff
@@ -664,10 +669,10 @@ def is_modified(self, modified_files: t.Optional[t.List[str]]) -> bool:
         if modified_files:
             for f in modified_files:
                 _f_fullpath = to_absolute_path(f)
-                if _f_fullpath.parts[-1].endswith('.md'):
+                if os.path.basename(_f_fullpath).endswith('.md'):
                     continue
 
-                if _app_dir_fullpath in _f_fullpath.parents:
+                if _app_dir_fullpath in os.path.join(_f_fullpath, '..'):
```

Comment:
the result of os.path.join('/a/b/c', '..') is '/a/b/c/..'. String `in` doesn't work well here.

Reply:
It is true, I forgot to use realpath in this line...

Reply:
Ooh, I see. In the previous commit, I used `realpath`, but I removed it because it was already `realpath`. All we do here is get the path to the parent from the current directory and perform the "in" action.

As I can see for example: `app_dir_full_path = /1/2/3, f_fullpath = /1/2/3/4`, it actually doesn't matter if it's the parent or not. We can directly check: `app_dir_full_path in f_fullpath`.

So in this commit, I just forgot to remove the join.

Reply:
I suggest replacing "app_dir_fullpath" with "startswith" in this context.